Assets/MixedRealityToolkit/_Core/Boundary/MixedRealityBoundarySystem.cs

```diff
@@ -66,35 +69,120 @@ public override void Reset()
+        {
+            return Contains(location, Boundary.Type.TrackedArea);
+        }
+
+        /// <inheritdoc/>
+        public bool Contains(Vector3 location, Boundary.Type boundaryType)
+        {
+            if (!EdgeUtils.IsValidPoint(location))
+            {
+                // Invalid location.
+                return false;
+            }
+
+            if (!FloorHeight.HasValue)
+            {
+                // No floor.
+                return false;
+            }
+
+            if ((FloorHeight.Value > location.y) ||
+                (BoundaryHeight < location.y))
+            {
+                // Location below the floor or above the boundary height.
+                return false;
+            }
+
+            // Boundary coordinates are always "on the floor"
+            Vector2 point = new Vector2(location.x, location.z);
+
+            if (boundaryType == Boundary.Type.PlayArea)
+            {
+                // Check the inscribed rectangle.
+                if (InscribedRectangularBounds != null)
+                {
+                    return InscribedRectangularBounds.IsInsideBoundary(point);
+                }
+            }
+            else if(boundaryType == Boundary.Type.TrackedArea)
+            {
+                // Check the geometry
+                return EdgeUtils.IsInsideBoundary(GeometryBounds, point);
+            }
+
+            // Not in either boundary type.
+            return false;
+        }
+
         /// <summary>
         /// Retrieves the boundary geometry and creates the boundary and inscribed playspace volumes.
         /// </summary>
         private void CalculateBoundaryBounds()
         {
+            // Reset the bounds
+            GeometryBounds = new Edge[0];
+            FloorHeight = null;
+            InscribedRectangularBounds = null;
+
+            // Boundaries are supported for Room Scale experiences only.
             if (XRDevice.GetTrackingSpaceType() != TrackingSpaceType.RoomScale)
             {
-                // Boundaries are supported for Room Scale experiences only.
                 return;
             }
 
-            OutscribedVolume = new Bounds();
+#if !UNITY_WSA
+            // This always returns false in Windows Mixed Reality.
+            if (!Boundary.configured)
+            {
+                Debug.Log("No boundary configured.");
+                return;
+            }
+#endif // !UNITY_WSA
 
             // Get the boundary geometry.
             List<Vector3> boundaryGeometry = new List<Vector3>(0);
-            if (Boundary.TryGetGeometry(boundaryGeometry))
+            List<Edge> boundaryEdges = new List<Edge>(0);
+
+            if (Boundary.TryGetGeometry(boundaryGeometry, Boundary.Type.TrackedArea))
             {
+                // FloorHeight starts out as null. Use a very high value for the floor to ensure
+                // that we do not accidentally set it too low.
+                float floorHeight = 1000f;
```

Comment:
Should this be configurable?

Reply:
It's just a high number (the floor should never be 1000m above the origin), don't think it really needs to be configurable. I *could* make the number bigger :)